duplicate_url_discarder/processors/subpath_removal.py

```diff
@@ -0,0 +1,37 @@
+import re
+from collections.abc import Iterable
+
+from scrapy.utils.url import parse_url
+
+from .base import UrlProcessorBase
+
+
+class SubpathRemovalProcessor(UrlProcessorBase):
+    """Removes the i-th subpath (starts at 0) from the URL path if it exists.
+
+    An example would be removing the 0-th and 2nd subpath of a URL like
+    https://example.com/a/b/c/d would result in https://example.com/b/c.
+    """
+
+    def validate_args(self) -> None:
+        base_exception_msg = "subpathRemoval args must be an iterable of integers."
+
+        if not isinstance(self.args, Iterable):
+            raise TypeError(
+                f"{base_exception_msg} Got type: {type(self.args)}: {self.args}."
+            )
+
+        for arg in self.args:
+            if not isinstance(arg, int):
+                raise ValueError(
+                    f"{base_exception_msg} Encountered {type(arg)}: {arg}."
+                )
+
+    def process(self, input_url: str) -> str:
+        positions = [int(pos) for pos in self.args]
```

Comment:
We already check that they are ints so is this needed?

Reply:
Good catch! This was from I was casting strings to integers but simply resorted to a stricter int as args.